utils/torch_solver.py

```diff
@@ -57,6 +57,13 @@ def get_next(stop_val):
         return stop_val + 1
 
     def run(self, callback):
+        # model weight initialization
+        self.model.initialize_model_weights()
```

Comment:
This is probably more expansive than the `clone` in `tf` model no?
Can't we simply create a copy of the model here?

Reply:
Maybe then rather than copy we should provide a `model_init_fn`, rather than a `model`. This allows us to get rid of potential sources of unfairness -> will try a proposal